File: grader_examples/template3lib.py
#!/usr/bin/env python
# template3lib.py
#
# Example external module dependency for template3
#
# The graders/ directory is added recursively to the path before running user
# submissions and solutions, so this module can be accessed at runtime by any
# standard "import" command. For example:
#
# from template3lib import add_one
# 
# Brian Moore
# brimoor@umich.edu
#
# January 6, 2017
# February 25, 2017
#

# Add one
def add_one(x):
    return x + 1

File: grader_examples/template3lib.m
%
% template3lib.m
%
% Example external module dependency for template3
%
% The graders/ directory is added recursively to the path before running
% user submissions and solutions, so this module can be accessed at runtime
% by calling the methods directory
% 
% NOTE: Here we mimic modules in Python/Julia by wrapping (static) methods
%       in a class, but one can simply include each function in separate
%       files (e.g., add_one.m) if desired
%
% Brian Moore
% brimoor@umich.edu
%
% January 6, 2017
% February 25, 2017
%
classdef template3lib
    methods (Access = public, Static = true)

        % Add one
        function y = add_one(x)
            y = x + 1;
        end

    end
end
